src/lib/variants.test.ts

```diff
@@ -162,4 +163,99 @@ describe('variant filters', () => {
         const got2 = excludeTags.test(test2, targeting2);
         expect(got2).toBe(false);
     });
+
+    it('should filter by max views', () => {
+        const viewLog = [
+            { date: new Date('2019-06-11T10:24:00').valueOf(), testId: 'example-1' },
+            { date: new Date('2019-07-11T10:24:00').valueOf(), testId: 'B' },
+            { date: new Date('2019-07-15T10:24:00').valueOf(), testId: 'example-1' },
+            { date: new Date('2019-07-17T10:24:00').valueOf(), testId: 'example-1' },
+            { date: new Date('2019-08-11T10:24:00').valueOf(), testId: 'example-1' },
+        ];
+
+        const now = new Date('2019-08-17T10:24:00');
+        const filter = withinMaxViews(viewLog, now);
+
+        const test1 = {
+            ...testDefault,
+            maxViews: {
+                maxViewsCount: 4,
+                maxViewsDays: 30,
+                minDaysBetweenViews: 0,
+            },
+            alwaysAsk: false,
+        };
+
+        const got1 = filter.test(test1, targetingDefault);
+
+        expect(got1).toBe(true);
+
+        const test2 = {
```

Comment:
What do you think about splitting this test in 2? Maybe `it("returns true when below the configured threshold" ...)` and `it("returns false when above the configured threshold" ...)`? I usually strive to have only 1 (or a few tightly coupled) assertions in the block, since I think shorter tests are usually easier to read.

Reply:
Yeah, I agree it is potentially confusing. I might try out my `toBeHint` helper as well - to use with a table here, as ideally I wanted a single test (`it(..`) per filter, but maybe that is not realistic.